src/main/java/net/quarrymod/block/QuarryBlock.java

```diff
@@ -35,16 +35,16 @@ protected void appendProperties(StateManager.Builder<Block, BlockState> builder)
     @Override
     public void onPlaced(World world, BlockPos pos, BlockState state, LivingEntity placer, ItemStack stack) {
         super.onPlaced(world, pos, state, placer, stack);
-        ((QuarryBlockEntity) world.getBlockEntity(pos)).resetOnPlaced();
+        if (world.getBlockEntity(pos) instanceof QuarryBlockEntity quarryEntity) {
+            quarryEntity.resetOnPlaced();
+        }
     }
 
     @Override
     public void onStateReplaced(BlockState state, World worldIn, BlockPos pos, BlockState newState, boolean isMoving) {
-        if (state.getBlock() != newState.getBlock()) {
-            QuarryBlockEntity quarryBlockEntity = ((QuarryBlockEntity) worldIn.getBlockEntity(pos));
-            if (quarryBlockEntity != null) {
-                ItemHandlerUtils.dropItemHandler(worldIn, pos, quarryBlockEntity.quarryUpgradesInventory);
-            }
+        if (state.getBlock() != newState.getBlock()
+            && worldIn.getBlockEntity(pos) instanceof QuarryBlockEntity quarryEntity) {
+            ItemHandlerUtils.dropItemHandler(worldIn, pos, quarryEntity.quarryUpgradesInventory);
```

Comment:
Dit even goed controleren, want als nu met een wrench dat ding weghaalt, behoud die zijn modules + settings. Kijken of dit straks nog werkt.